spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -822,91 +1214,30 @@ def __write_block(
                     weight_scales, row_data, delayed_row_data,
                     machine_time_step))
                 conn_holder.finish()
 
-        index = None
-        if row_data.size:
-            block_addr, single_addr, index = self.__write_row_data(
-                spec, synapse_info.connector, pre_vertex_slice,
-                post_vertex_slice, row_length, row_data, rinfo,
-                single_synapses, synaptic_matrix_region, block_addr,
-                single_addr, app_edge, synapse_info)
-        elif rinfo is not None:
-            index = self.__poptable_type.update_master_population_table(
-                0, 0, rinfo.first_key_and_mask)
-        del row_data
-
-        if block_addr > all_syn_block_sz:
-            raise Exception(
-                "Too much synaptic memory has been written: {} of {} ".format(
-                    block_addr, all_syn_block_sz))
-
-        delay_rinfo = None
-        delay_key = (app_edge.pre_vertex, pre_vertex_slice.lo_atom,
-                     pre_vertex_slice.hi_atom)
-        if delay_key in self.__delay_key_index:
-            delay_rinfo = self.__delay_key_index[delay_key]
-        d_index = None
-        if delayed_row_data.size:
-            block_addr, single_addr, d_index = self.__write_row_data(
-                spec, synapse_info.connector, pre_vertex_slice,
-                post_vertex_slice, delayed_row_length, delayed_row_data,
-                delay_rinfo, single_synapses, synaptic_matrix_region,
-                block_addr, single_addr, app_edge, synapse_info)
-        elif delay_rinfo is not None:
-            d_index = self.__poptable_type.update_master_population_table(
-                0, 0, delay_rinfo.first_key_and_mask)
-        del delayed_row_data
-
-        if block_addr > all_syn_block_sz:
-            raise Exception(
-                "Too much synaptic memory has been written: {} of {} ".format(
-                    block_addr, all_syn_block_sz))
-        if d_index is not None and index is not None and index != d_index:
-            raise Exception(
-                "Delay index {} and normal index {} do not match".format(
-                    d_index, index))
-        return block_addr, single_addr, index
+        return (row_data, delayed_row_data)
 
     def __is_direct(
-            self, single_addr, connector, pre_vertex_slice, post_vertex_slice,
-            app_edge, synapse_info):
+            self, single_addr, s_info, pre_vertex_slice, post_vertex_slice,
+            is_delayed):
         """ Determine if the given connection can be done with a "direct"\
             synaptic matrix - this must have an exactly 1 entry per row
         """
         return (
-            app_edge.n_delay_stages == 0 and
-            connector.use_direct_matrix(synapse_info) and
-            (single_addr + (pre_vertex_slice.n_atoms * BYTES_PER_WORD) <=
+            not is_delayed and
+            isinstance(s_info.connector, OneToOneConnector) and
```

Comment:
I'm reasonably sure that this is the only use of connector.use_direct_matrix(...) that's been deleted here, so it can probably now be removed from the AbstractConnector and OneToOneConnector.

Reply:
Ah yes, this seems to have gone elsewhere now.